### PR TITLE
Sitemap fixes

### PR DESCRIPTION
**Pull request by thebetauser**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

---

These changes fix multiple sitemap issues including:
* Current non-working implementation
* Support for parsing multiple urls in the sitemap index file
* Missing DataType.SITEMAP class in data_formatter.py 

---

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (does not change functionality, e.g. code style improvements, linting)
- [ ] Documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Please delete options that are not relevant.

- [ ] Unit Test
- [ ] Test Script (please provide)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Made sure Checks passed

### CODE REVIEW

**Comment by cachho** [comment]
Thanks. So it recursively loads sitemap indices now?

**Comment by ipsoftar** [comment]
> Thanks. So it recursively loads sitemap indices now?

The original code indexed all the urls listed in the sitemap. 

This PR fixed what appears to be a coding error in the return statement which returns only index 0 of the list instead of the full list. 

It also fixed an attribute error when parsing the web_page_loader data

**Comment by thebetauser** [comment]
> Thanks. So it recursively loads sitemap indices now?

The original code indexed all the urls listed in the sitemap.

This PR fixed what appears to be a coding error in the return statement which returns only index 0 of the list instead of the full list.

It also fixed an attribute error when parsing the web_page_loader data

**Comment by cachho** [comment]
> If there is a need for it, I can work on making the function recursive.
It would certainly be useful. But please do yourself the favor and research this before you start, sitemap indices aren't just sitemaps that use sitemaps as locations, there are a few differences to clearly tell them apart. Thanks

File: pyproject.toml [hunk 1]
```diff
@@ -91,6 +91,7 @@ youtube-transcript-api = "^0.6.1"
 beautifulsoup4 = "^4.12.2"
 pypdf = "^3.11.0"
 pytube = "^15.0.0"
+lxml = "^4.6.3" # required for sitemap xml parsing
```
Comment (cachho) [inline pyproject.toml:94]: How come this dependency is needed all of the sudden? I don't see you import it.

Comment (thebetauser) [reply pyproject.toml:94]: This is needed to allow beautiful soup to parse xml formatted files. Some sitemap.xml files are in xml format while others are in html format.

I was hoping this was going to be a quick fix but it seems that the sitemap implementation for this project is not complete. I'm going to pull this PR and submit it once I have a working implementation.

**Pull request closed without merging**